### ISSUE TITLE
Unify handling of CLC snippets on bare-metal platform

### ISSUE THREAD

**Issue by invidian** [kinvolk/lokomotive#442]
All platforms expose `controller_clc_snippets` and `clc_snippets` per worker pool. On bare metal we don't allow any CLC snippets and existing, underlying Terraform code we have for handling them also differs a lot from other platforms.

**Comment by invidian** [comment]
@ipochi I think this is now addressed? Except that on bare metal we don't have worker pools, so we only have `clc_snippets`, right?

**Comment by ipochi** [comment]
Yes. 
Baremetal platforms don't support the concept of worker pools. Although it would be nice to have.

W.r.t to clc snippets, baremetal platform makes up slightly by having the ability of different clc snippet per node.

**Comment by invidian** [comment]
Right. Hmm, I wonder if we should finish cleaning it up together with #443 or separately, as a preparation for #443...
